src/dotenv/cli.py

```diff
@@ -44,56 +48,70 @@ def cli(ctx: click.Context, file: Any, quote: Any, export: Any) -> None:
                    "which displays name=value without quotes.")
 def list(ctx: click.Context, format: bool) -> None:
     '''Display all the stored key/value.'''
-    file = ctx.obj['FILE']
-    if not os.path.isfile(file):
-        raise click.BadParameter(
-            'Path "%s" does not exist.' % (file),
-            ctx=ctx
-        )
-    dotenv_as_dict = dotenv_values(file)
+    
+    d = {}
+    for file in ctx.obj['FILES']:
+        if not os.path.isfile(file):
+            raise click.BadParameter(
+                'Path "%s" does not exist.' % (file),
+                ctx=ctx
+            )
+        dotenv_as_dict = dotenv_values(file)
+
+        for k, v in dotenv_as_dict.items():
+            d[k] = v
+
     if format == 'json':
-        click.echo(json.dumps(dotenv_as_dict, indent=2, sort_keys=True))
+        click.echo(json.dumps(d, indent=2, sort_keys=True))
     else:
         prefix = 'export ' if format == 'export' else ''
-        for k in sorted(dotenv_as_dict):
-            v = dotenv_as_dict[k]
+        for k in sorted(d):
+            v = d[k]
             if v is not None:
                 if format in ('export', 'shell'):
                     v = shlex.quote(v)
                 click.echo('%s%s=%s' % (prefix, k, v))
-
+        
 
 @cli.command()
 @click.pass_context
 @click.argument('key', required=True)
 @click.argument('value', required=True)
 def set(ctx: click.Context, key: Any, value: Any) -> None:
     '''Store the given key/value.'''
-    file = ctx.obj['FILE']
+    file = ctx.obj['FILES']
     quote = ctx.obj['QUOTE']
     export = ctx.obj['EXPORT']
-    success, key, value = set_key(file, key, value, quote, export)
-    if success:
-        click.echo('%s=%s' % (key, value))
-    else:
-        exit(1)
+    for file in ctx.obj['FILES']:
+        success, key, value = set_key(file, key, value, quote, export)
+        if success:
+            click.echo('%s=%s' % (key, value))
+        else:
+            exit(1)
 
 
 @cli.command()
 @click.pass_context
 @click.argument('key', required=True)
 def get(ctx: click.Context, key: Any) -> None:
     '''Retrieve the value for the given key.'''
-    file = ctx.obj['FILE']
-    if not os.path.isfile(file):
-        raise click.BadParameter(
-            'Path "%s" does not exist.' % (file),
-            ctx=ctx
-        )
-    stored_value = get_key(file, key)
-    if stored_value:
-        click.echo(stored_value)
+    
+    value, set = None, False
+    files = ctx.obj['FILES']
+    for file in files:
+        if not os.path.isfile(file):
+            raise click.BadParameter(
+                'Path "%s" does not exist.' % (file),
+                ctx=ctx
+            )
+        stored_value = get_key(file, key)
+        if stored_value:
+            value, set = stored_value, True
+    
+    if set:
+        click.echo(value)
     else:
+        logger.warning(f"Key {key} not found in {files[0] if len(files) == 1 else files}.")
```

Comment:
Moved warning to cli: key may not be present in one file, but present in another.

Not sure how this will affect other calls to this function.

Reply:
Looks a bit strange right now:

```
python3 dotenv/ -f test1.env -f test2.env get URLa
Key URLa not found in test1.env.
Key URLa not found in test2.env.
Key URLa not found in ('test1.env', 'test2.env').
```

Reply:
Fixed by [f730c6d](https://github.com/theskumar/python-dotenv/pull/424/commits/f730c6d92900f8741ae03f24e177b1a3ef32ecfc)